src/converters/app-converter.ts

```diff
@@ -0,0 +1,388 @@
+            }
+
+            case 'Fn::Join':
+                return lift(([delim, strings]) => strings.join(delim), this.processIntrinsics(params));
+
+            case 'Fn::Select':
+                return lift(([index, list]) => list[index], this.processIntrinsics(params));
+
+            case 'Fn::Split':
+                return lift(([delim, str]) => str.split(delim), this.processIntrinsics(params));
+
+            case 'Fn::Base64':
+                return lift((str) => Buffer.from(str).toString('base64'), this.processIntrinsics(params));
+
+            case 'Fn::Cidr':
+                return lift(
+                    ([ipBlock, count, cidrBits]) =>
+                        cidr({
+                            ipBlock,
+                            count,
+                            cidrBits,
+                        }).then((r) => r.subnets),
+                    this.processIntrinsics(params),
+                );
+
+            case 'Fn::GetAZs':
+                return lift(([region]) => getAzs({ region }).then((r) => r.azs), this.processIntrinsics(params));
+
+            case 'Fn::Sub':
+                return lift((params) => {
+                    const [template, vars] =
+                        typeof params === 'string' ? [params, undefined] : [params[0] as string, params[1]];
+
+                    const parts: string[] = [];
+                    for (const part of parseSub(template)) {
+                        parts.push(part.str);
+
+                        if (part.ref !== undefined) {
+                            if (part.ref.attr !== undefined) {
+                                parts.push(this.resolveAtt(part.ref.id, part.ref.attr!));
+                            } else {
+                                parts.push(this.resolveRef(part.ref.id));
+                            }
+                        }
+                    }
+
+                    return lift((parts) => parts.map((v: any) => v.toString()).join(''), parts);
+                }, this.processIntrinsics(params));
+
+            case 'Fn::Transform': {
+                // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-macros.html
+                throw new Error('Fn::Transform is not supported – Cfn Template Macros are not supported yet');
+            }
+
+            case 'Fn::ImportValue': {
+                // TODO: support cross cfn stack references?
+                // This is related to the Export Name from outputs https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html
+                // We might revisit this once the CDKTF supports cross stack references
+                throw new Error(`Fn::ImportValue is not yet supported.`);
+            }
+
+            default:
+                throw new Error(`unsupported intrinsic function ${fn} (params: ${JSON.stringify(params)})`);
+        }
+    }
+
+    private resolveRef(target: any): any {
```

Comment:
This looks like it's emitting Pulumi data source calls?

Reply:
Yes it does. Some builtin CloudFormation intrinsics are akin to data source calls. I should better document this.